### PR TITLE
Feature: Support Custom statuses code

### LINKED ISSUE #1117: Support custom `Status`es for `HttpError`
**Is your feature request related to a problem? Please describe.**
I would like to create an `HttpError` with a custom status [451 Unavailable For Legal Reasons](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/451).

**Describe the solution you'd like**
A way to create a custom `Status` instance for my `HttpError`.

### PR DESCRIPTION
**Pull request by gciuloaica**
fixes #1117

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/http/HttpError.scala [hunk 1]
```diff
@@ -121,4 +122,6 @@ object HttpError {
 
   final case class BadGateway(msg: String = "Bad Gateway") extends HttpError(Status.BAD_GATEWAY, msg)
 
+  final case class CustomResponseStatus(code: Int, reason: String) extends HttpError(CUSTOM_STATUS(code), reason)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/HttpError.scala]:
```suggestion
  final case class Custom(code: Int, reason: String) extends HttpError(Status.Custom(code), reason)
```

File: zio-http/src/main/scala/zhttp/http/Status.scala [hunk 2]
```diff
@@ -90,62 +91,63 @@ sealed trait Status extends Product with Serializable { self =>
 }
 
 object Status {
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/Status.scala]: All Statuses should be in `CamelCase`. May be a separate PR for this?

File: zio-http/src/main/scala/zhttp/http/HttpError.scala (hunk 1 above)
Comment (gciuloaica) [reply zio-http/src/main/scala/zhttp/http/HttpError.scala]: done

File: zio-http/src/main/scala/zhttp/http/HttpError.scala [hunk 3]
```diff
@@ -121,4 +121,6 @@ object HttpError {
 
   final case class BadGateway(msg: String = "Bad Gateway") extends HttpError(Status.BAD_GATEWAY, msg)
 
+  final case class CustomResponseStatus(code: Int, reason: String) extends HttpError(Status.Custom(code), reason)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/HttpError.scala]:
```suggestion
  final case class Custom(code: Int, reason: String) extends HttpError(Status.Custom(code), reason)
```

**Review by tusharmath — APPROVED** [review]
